### PR TITLE
build out the interface definitions for validator keys api

### PR DESCRIPTION
**Pull request by rolfyone**
Pushing our dev interface to be closer to the newly published swagger for validator keys api.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java [hunk 1]
```diff
@@ -13,12 +13,73 @@
...
+   * @return The result of each deletion, and slashing protection data
+   */
+  public DeleteKeysResponse deleteValidators(final List<BLSPublicKey> validators) {
+    throw new NotImplementedException("deleteValidators not implemented yet");
+  }
+
+  /**
+   * import a collection of validators The result needs to be symmetric with the input order Must *
+   * supply a message in case of error. Super important that an individual error doesn't fail the *
+   * whole process, each should be its own process that can error out.
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java]: Not a big deal but the formatting on this one didn't work out (and should start with a capital I).

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/ValidatorTypes.java [hunk 2]
```diff
@@ -21,12 +21,55 @@
...
+import tech.pegasys.teku.validator.client.restapi.apis.schema.PostKeysRequest;
 
 public class ValidatorTypes {
+
+  static SerializableTypeDefinition<PostKeyResult> POST_KEY_RESULT =
+      SerializableTypeDefinition.<PostKeyResult>object()
+          .name("PostKeyResult")
+          .withField(
+              "status",
+              DeserializableTypeDefinition.enumOf(ImportStatus.class),
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/ValidatorTypes.java]: I wonder if we should use a static import for `enumOf` (and probably `listOf` elsewhere).  May make this read better. Actually even `STRING_TYPE` might be better as a static import.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/PostKeys.java [hunk 3]
```diff
@@ -30,13 +32,32 @@ public PostKeys() {
...
+    //    } else if (body.getKeystores().size() != body.getPasswords().size()) {
+    //      request.respondError(
+    //          SC_BAD_REQUEST, "Different number of passwords and keystores supplied, cannot
+    // process.");
+    //      return;
+    //    }
+
+    // TODO call keyManager.importValidators(body.getKeystores(), body.getPasswords(),
+    // body.getSlashingProtection());
+    // and set request.respondOk if it doesnt fail  - it shouldn't fail...
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/PostKeys.java]: Also not a big deal but I'd probably remove the commented out code.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/schema/PostKeyResult.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
+import java.util.Optional;
+
+public class PostKeyResult {
+  private final ImportStatus importStatus;
+  private final Optional<String> message;
+
+  PostKeyResult(final ImportStatus importStatus, final Optional<String> message) {
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/schema/PostKeyResult.java]: Any reason not to make this private?

**Review by ajsutton — APPROVED** [review]
LGTM.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java [hunk 5]
```diff
@@ -13,12 +13,73 @@
...
+import tech.pegasys.teku.validator.client.restapi.apis.schema.ImportStatus;
 
 public class KeyManager {
 
+  /**
+   * Get a listing of active validator keys
+   *
+   * @return a list of active validators
+   */
+  List<ActiveValidator> getActiveValidatorKeys() {
```
Comment (regio) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java:32]: Should we set this method as public?

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/PostKeys.java (hunk 3 above)
Comment (rolfyone) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/PostKeys.java]: can do, its just basically some of the error checking and was breaking a test...

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java (hunk 5 above)
Comment (rolfyone) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java:32]: copied from interface, so yep probably shoudl be

**Pull request merged**